File: wfe-restapi/src/main/java/ru/runa/wfe/rest/dto/WfProcessDto.java
package ru.runa.wfe.rest.dto;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import lombok.Data;

@Data
public class WfProcessDto {
    private Long id;
    private String definitionName;
    private String executionStatus;
    private Date startDate;
    private Date endDate; 
    private List<WfVariableDto> variables = new ArrayList<>();
}

File: wfe-spa/src/ts/WfProcessDto.ts
import { WfVariableDto } from './WfVariableDto';

export class WfProcessDto {
    id: number|null = null;
    definitionName: string = '';
    executionStatus: string = '';
    startDate: Date|null = null;
    endDate: Date|null = null; 
    variables: Array<WfVariableDto>|null = null;

    getInfo(): object {
        return {
            'Номер': this.id,
            'Имя процесса': this.definitionName,
            'Запущен': this.startDate ? new Date(this.startDate).toLocaleString() : '',
            'Завершён': this.endDate ? new Date(this.endDate).toLocaleString() : '',
            'Статус': this.executionStatus 
        };
    }
}